### PR TITLE
add prompt_for_turn, fix Turn.toJSON()

### LINKED ISSUE #9: Some way to let users take turns without seeing their opponent's turn
The best way to do this would probably be through DMs

### PR DESCRIPTION
**Pull request by dyc3**
closes #9

Test with `p!turn`﻿

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -35,8 +36,373 @@ async def challenge(ctx: commands.Context, opponent: str):
...
+						"Priority": 0,
+						"Power": 75,
+						"Accuracy": 90
+					}],
+					"Friendship": 0,
+					"OriginalTrainerID": 0,
+					"Type": 0
+				}
+			}]
+		})
```
Comment (elenirotsides) [inline bot.py:375]: Where is all this data coming from and what does it mean? Is this supposed to be placeholder data for now that's actually supposed to be coming from the API?

Comment (dyc3) [reply bot.py:375]: Yes, it's placeholder.

File: bot.py [hunk 2]
```diff
@@ -35,8 +36,373 @@ async def challenge(ctx: commands.Context, opponent: str):
...
+	for move in battlecontext["Pokemon"]["Moves"]:
+		embed.add_field(name=move['Name'], value=f"{move['Type']} {move['CurrentPP']} {move['MaxPP']}")
+	msg: Message = await user.dm_channel.send(content="Select a move", embed=embed)
+	for i in RESPONSE_REACTIONS:
+		await msg.add_reaction(i)
+	def check(payload):
+		return payload.message_id == msg.id and payload.user_id == user.id and str(payload.emoji) in RESPONSE_REACTIONS
+	try:
+		# HACK: reaction_add doesn't work in DMs
+		payload = await bot.wait_for("raw_reaction_add", timeout=10, check=check)
```
Comment (elenirotsides) [inline bot.py:396]:
```suggestion
		payload = await bot.wait_for("raw_reaction_add", timeout=20, check=check)
```

Comment (elenirotsides) [reply bot.py:396]: @dyc3 It timed out too fast in my opinion; by the time I got to my dm's the timeout had hit

Comment (dyc3) [reply bot.py:396]: whoops meant to remove this

**Review by elenirotsides — APPROVED** [review]

**Pull request merged**